eclipse/build.gradle.kts

```diff
@@ -58,4 +63,131 @@ tasks {
     artifacts {
         add("testing", testJar)
     }
+
+
+    /* Eclipse release tasks
+     *
+     * The following tasks provide the functionality of creating
+     * an eclipse update-site (via "updateSite") or dropin (via "dropin").
+     * The creation of the update-site uses as recommended the eclipse's
+     * P2Directory. You can find a how-to here:
+     * http://maksim.sorokin.dk/it/2010/11/26/creating-a-p2-repository-from-features-and-plugins/
+     *
+     * Instead of calling the P2Director via CLI we use the goomph
+     * plugin's classes that provide an abstraction layer of the P2Director.
+     */
+
+    val updateSiteDirPath = "build/update-site"
+
+    /* Step 0 of update-site creation
+     *
+     * Creates the structure:
+     * update-site/
+     *  |- features/
+     *    |- feature.xml
+     *  |- plugins/
+     *    |- saros.core.jar
+     *    |- saros.eclipse.jar
+     *  |- category.xml
+     */
+    val updateSitePreparation by registering(Copy::class) {
+        dependsOn(":saros.core:jar", ":saros.eclipse:jar")
+
+        from("feature/feature.xml") {
+            into("features")
+        }
+        into("plugins") {
+            from(project.tasks.findByName("jar"))
+            from(project(":saros.core").tasks.findByName("jar"))
+        }
+        from("update/site.xml")
+        into(updateSiteDirPath)
+    }
+
+    /* Step 1 of update-site creation
+     *
+     * Creates the basic p2-Repository, but it is
+     * not usable as update-site, because the plugins
+     * are not visible to users.
+     *
+     * equivalent to P2Director call with:
+     * <code>
+     * org.eclipse.equinox.p2.publisher.FeaturesAndBundlesPublisher \
+     *   -source <project dir>/build/update-site \
+     *   -compress \
+     *   -inplace \
+     *   -append \
+     *   -publishArtifacts
+     * </code>
+     */
+    val updateSiteFeaturesAndBundlesPublishing by registering {
+        dependsOn(updateSitePreparation)
+        doLast {
+            with(com.diffplug.gradle.p2.FeaturesAndBundlesPublisher()) {
+                source(project.file(updateSiteDirPath))
+                compress()
+                inplace()
+                append()
+                publishArtifacts()
+                runUsingBootstrapper()
+            }
+        }
+    }
+
+    /* Step 2 of update-site creation
+     *
+     * Adds the meta-data to the repository to make
+     * the plugin accessible for users.
+     *
+     * equivalent to P2Director call with:
+     * <code>
+     * org.eclipse.equinox.p2.publisher.CategoryPublisher \
+     *   -metadataRepository file:<project dir>/build/update-site \
+     *   -categoryDefinition file:<project dir>/build/update-site/site.xml
+     * </code>
+     */
+    val updateSiteCategoryPublishing by registering {
+        dependsOn(updateSiteFeaturesAndBundlesPublishing)
+        doLast {
+            with(saros.gradle.eclipse.CategoryPublisher("4.8.0")) {
```

Comment:
Is this the specification of the targeted Eclipse release? If so, is there a way of unifying it with the declaration in [`SarosEclipseExtension`](https://github.com/saros-project/saros/blob/cbe1fe64664090fc2ce0a88d450e37b852a26b0f/buildSrc/src/main/java/saros/gradle/eclipse/SarosEclipseExtension.java#L27) so that we don't have to specify the version in two different places?